### DISCUSSION TITLE
Blob read 0 magic header bytes , Image Exception

### DISCUSSION THREAD

**Discussion by programmerarunkumar** [ImageMagick/ImageMagick#2609] [Help]
<event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Wand Magick[1190]: magick-wand.c/NewMagickWand/1080/Wand
  MagickWand-8</event>
  <event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Wand Magick[1190]: magick-image.c/MagickReadImage/8801/Wand
  MagickWand-8</event>
  <event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Policy Magick[1190]: policy.c/IsRightsAuthorized/611/Policy
  Domain: Path; rights=Read; pattern="OriginalImages/Image17.jpeg" ...</event>
  <event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Blob Magick[1190]: blob.c/OpenBlob/3390/Blob
    read 0 magic header bytes</event>
  <event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Cache Magick[1190]: cache.c/DestroyPixelCache/1041/Cache
  destroy </event>
  <event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Policy Magick[1190]: policy.c/IsRightsAuthorized/611/Policy
  Domain: Path; rights=Read; pattern="OriginalImages/Image17.jpeg" ...</event>
  <event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Blob Magick[1190]: blob.c/OpenBlob/3390/Blob
    read 0 magic header bytes</event>
  <event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Cache Magick[1190]: cache.c/DestroyPixelCache/1041/Cache
  destroy </event>
  <event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Policy Magick[1190]: policy.c/IsRightsAuthorized/611/Policy
  Domain: Coder; rights=Read; pattern="JPEG" ...</event>
  <event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Policy Magick[1190]: policy.c/IsRightsAuthorized/611/Policy
  Domain: Path; rights=Read; pattern="OriginalImages/Image17.jpeg" ...</event>
  <event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Blob Magick[1190]: blob.c/OpenBlob/3390/Blob
    read 0 magic header bytes</event>
  <event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Locale Magick[1190]: locale.c/GetLocaleOptions/854/Locale
  Searching for locale file: "/usr/local/share/ImageMagick-7/locale.xml"</event>
  <event>2020-09-23T10:33:54+00:00 0:10.729 21.380u 7.0.9 Locale Magick[1190]: locale.c/GetLocaleOptions/854/Locale
  Searching for locale file: "/usr/local/lib/ImageMagick-7.0.9//config-Q16HDRI/locale.xml"</event>
  <event>2020-09-23T10:33:54+00:00 0:10.730 21.380u 7.0.9 Locale Magick[1190]: locale.c/GetLocaleOptions/854/Locale
  Searching for locale file: "/usr/local/etc/ImageMagick-7/locale.xml"</event>
  <event>2020-09-23T10:33:54+00:00 0:10.730 21.380u 7.0.9 Locale Magick[1190]: locale.c/GetLocaleOptions/854/Locale
  Searching for locale file: "/usr/local/share/doc/ImageMagick-7/locale.xml"</event>
  <event>2020-09-23T10:33:54+00:00 0:10.730 21.380u 7.0.9 Locale Magick[1190]: locale.c/GetLocaleOptions/854/Locale
  Searching for locale file: "arun/.config/ImageMagick/locale.xml"</event>
  <event>2020-09-23T10:33:54+00:00 0:10.730 21.380u 7.0.9 Configure Magick[1190]: locale.c/LoadLocaleCache/1206/Configure
  Loading locale configure file "/usr/local/share/ImageMagick-7/locale.xml" ...</event>
  <event>2020-09-23T10:33:54+00:00 0:10.730 21.380u 7.0.9 Locale Magick[1190]: locale.c/GetLocaleOptions/854/Locale
  Searching for locale file: "/usr/local/share/ImageMagick-7/english.xml"</event>
  <event>2020-09-23T10:33:54+00:00 0:10.730 21.380u 7.0.9 Locale Magick[1190]: locale.c/GetLocaleOptions/854/Locale
  Searching for locale file: "/usr/local/lib/ImageMagick-7.0.9//config-Q16HDRI/english.xml"</event>
  <event>2020-09-23T10:33:54+00:00 0:10.730 21.380u 7.0.9 Locale Magick[1190]: locale.c/GetLocaleOptions/854/Locale
  Searching for locale file: "/usr/local/etc/ImageMagick-7/english.xml"</event>
  <event>2020-09-23T10:33:54+00:00 0:10.730 21.380u 7.0.9 Locale Magick[1190]: locale.c/GetLocaleOptions/854/Locale
  Searching for locale file: "/usr/local/share/doc/ImageMagick-7/english.xml"</event>
  <event>2020-09-23T10:33:54+00:00 0:10.730 21.380u 7.0.9 Locale Magick[1190]: locale.c/GetLocaleOptions/854/Locale
  Searching for locale file: "/home/local/arun/.config/ImageMagick/english.xml"</event>
  <event>2020-09-23T10:33:54+00:00 0:10.730 21.380u 7.0.9 Configure Magick[1190]: locale.c/LoadLocaleCache/1206/Configure
  Loading locale configure file "/usr/local/share/ImageMagick-7/english.xml" ...</event>
  <event>2020-09-23T10:33:54+00:00 0:10.731 21.380u 7.0.9 Exception Magick[1190]: jpeg.c/ReadJPEGImage_/1191/Exception
  insufficient image data in file `OriginalImages/Image17.jpeg'</event>
  <event>2020-09-23T10:33:54+00:00 0:10.731 21.380u 7.0.9 Cache Magick[1190]: cache.c/DestroyPixelCache/1041/Cache
  destroy </event>
  <event>2020-09-23T10:33:54+00:00 0:10.731 21.380u 7.0.9 Wand Magick[1190]: magick-image.c/MagickResizeImage/9143/Wand
  MagickWand-8</event>
  <event>2020-09-23T10:33:54+00:00 0:10.731 21.380u 7.0.9 Exception Magick[1190]: magick-image.c/MagickResizeImage/9145/Exception
  wand contains no images `MagickWand-8'</event>
  <event>2020-09-23T10:33:54+00:00 0:10.731 21.380u 7.0.9 Wand Magick[1190]: magick-image.c/MagickWriteImage/13082/Wand
  MagickWand-8</event>
  <event>2020-09-23T10:33:54+00:00 0:10.731 21.380u 7.0.9 Exception Magick[1190]: magick-image.c/MagickWriteImage/13084/Exception
  wand contains no images `MagickWand-8'</event>
  <event>2020-09-23T10:33:54+00:00 0:10.731 21.380u 7.0.9 Wand Magick[1190]: magick-wand.c/DestroyMagickWand/171/Wand
  MagickWand-8</event>

why this exception occuring ?
There is no problem with my image

**Comment by snibgo** [comment]
What version of IM? On what platform?

What was your command?

Please post a link to your input file.

**Reply by programmerarunkumar** [reply]
> What version of IM? On what platform?
> 
> What was your command?
> 
> Please post a link to your input file.

  NewMagickWand();
  MagickReadImage();
  MagickResizeImage();
  MagickWriteImage();
  DestroyMagickWand();

Im using these MagickWand method to resize the image In C language & calling this C resize code from Java using JNA.


ImageMagick 7.0.9-21

Ubuntu

**Reply by fmw42** [reply]
You have not answered the questions

**Reply by programmerarunkumar** [reply]
> Please post a link to your input file.

Everytime Im getting for different file

**Comment by snibgo** [comment]
For us to help you, we need to reproduce the problem. We can't reproduce it from what you have said.

Those MagickWand functions take parameters, but you don't show any.

Please paste a complete but minimal program that has the problem, and a sample input file.

**Reply by programmerarunkumar** [reply]
> For us to help you, we need to reproduce the problem. We can't reproduce it from what you have said.
> 
> Those MagickWand functions take parameters, but you don't show any.
> 
> Please paste a complete but minimal program that has the problem, and a sample input file.



https://drive.google.com/drive/folders/1gC7RcMqMm50ekkU4cgA0TiaYvBmPf9qm?usp=sharing

Follow the Steps in the CheckList.txt to reproduce the issue

**Reply by programmerarunkumar** [reply]
> > For us to help you, we need to reproduce the problem. We can't reproduce it from what you have said.
> > Those MagickWand functions take parameters, but you don't show any.
> > Please paste a complete but minimal program that has the problem, and a sample input file.
> 
> https://drive.google.com/drive/folders/1gC7RcMqMm50ekkU4cgA0TiaYvBmPf9qm?usp=sharing
> 
> Follow the Steps in the CheckList.txt to reproduce the issue

Use the above my code and images to reproduce 
some of the image is corrupting
help me to solve this issue

**Reply by programmerarunkumar** [reply]
> > For us to help you, we need to reproduce the problem. We can't reproduce it from what you have said.
> > Those MagickWand functions take parameters, but you don't show any.
> > Please paste a complete but minimal program that has the problem, and a sample input file.
> 
> https://drive.google.com/drive/folders/1gC7RcMqMm50ekkU4cgA0TiaYvBmPf9qm?usp=sharing
> 
> Follow the Steps in the CheckList.txt to reproduce the issue
Please do like this

**Comment by snibgo** [comment]
Your resize.c has no error catching or reporting, so if MagickReadImage() fails you still try to resize it, etc. The function resize() always returns "done" even if there has been a problem.

How large are your images, in pixels?

The OP error is with OriginalImages/Image17.jpeg, but you haven't shown that file. Does the problem always occur with that file?

I can't comment on the java code. If you make that single-threaded, is there still a problem?

**Reply by programmerarunkumar** [reply]
> I can't comment on the java code. If you make that single-threaded, is there still a problem?

Using synchronization concept in java 
Synchronization  guarantees that no two threads can execute a synchronized method
Im not executing Multithread concept in my code

**Reply by programmerarunkumar** [reply]
> Your resize.c has no error catching or reporting, so if MagickReadImage() fails you still try to resize it, etc. The function resize() always returns "done" even if there has been a problem.
> 
> How large are your images, in pixels?
> 
> The OP error is with OriginalImages/Image17.jpeg, but you haven't shown that file. Does the problem always occur with that file?

I have shared my code and images  . Try with the help of CheckList.txt

**Reply by programmerarunkumar** [reply]
> > Your resize.c has no error catching or reporting, so if MagickReadImage() fails you still try to resize it, etc. The function resize() always returns "done" even if there has been a problem.
> > How large are your images, in pixels?
> > The OP error is with OriginalImages/Image17.jpeg, but you haven't shown that file. Does the problem always occur with that file?
> 
> I have shared my code and images . Try with the help of CheckList.txt

Can you Help me to resolve this issue

**Comment by snibgo** [comment]
I suggest you change resize.c to test for errors.

I suggest you change your java code to make it single-threaded.

How large are your images, in pixels?

The OP error is with OriginalImages/Image17.jpeg. Please link to that file. Does the problem always occur with that file?

**Reply by programmerarunkumar** [reply]
> I suggest you change your java code to make it single-threaded.

Im already replied for this question that my java code is synchronized and only one image will be resized at time

**Reply by programmerarunkumar** [reply]
> The OP error is with OriginalImages/Image17.jpeg. Please link to that file. Does the problem always occur with that file?

Every Time Im getting error for Different Image not only with OriginalImages/Image17.jpeg
